src/crystal/system/unix/syscall.cr

```diff
@@ -4,5 +4,12 @@ require "c/unistd"
 require "syscall"
 
 module Crystal::System::Syscall
-  ::Syscall.def_syscall getrandom, LibC::SSizeT, buf : UInt8*, buflen : LibC::SizeT, flags : UInt32
+  ::Syscall.def_syscall getrandom, LibC::SSizeT, buf : UInt8*, buflen : LibC::SizeT, flags : GetRandomFlags
+
+  @[Flags]
+  enum GetRandomFlags : UInt32
+    NonBlock # Don't block and return EAGAIN instead
+    Random   # No effect
+    Insecure # Return non-cryptographic random bytes
+  end
```

Comment:
This looks over-engineered. We shouldn't need an enum for such a low-level internal API. The syscall is not expected to be used outside our internal implementation. The parameter type could be just `UInt32` and we use the respective value at the call site, maybe provide the respective constants. But an enum seems over-kill for this.

Reply:
Isn't it a bit much calling the use of a  3 value enum over-engineered? It is not as if it adds and maintenance burden, or reduces the clarity of the code. IMO both variants would be fine.

Reply:
All relevant use cases (which is only a single one) needs only one of these values. There is no benefit in defining the other, unused ones as well. Let alone introducing an entire type for that. 

Maintenance and clarity may not be a huge concern. But it's still more code than necessary, so it's more to read and more to understand.

Besides that, there is also a performance implication. Every additional type puts more load on the compiler. Even though the effect is minimal, it's completely avoidable for zero loss. `Random` is part of the core lib, so it affects *every* Crystal program.

Reply:
I believe an enum is much more clear and maintainable than a literal number at the call site with a comment explaining it. This is the exact use case for enums.

Is declaring it really measurable in terms of compiler performance? (because if it is, then we should focus on optimizing enum creation on the compiler...)

Regardless, I'm changing it as requested.

Reply:
Constants are okay:

```crystal
lib LibC
  GRND_NONBLOCK = 1
end
```

What the comment means is that we are doing this in the `LibC` bindings:

```crystal
# dirent.cr

lib LibC
  DT_UNKNOWN =  0
  DT_DIR     =  4
  DT_LINK    = 10
end
```

instead of:

```crystal
lib LibC
  enum DirentType
    DT_BLK     = ...
    DT_CHR     = ...
    DT_DIR     = 4
    DT_FIFO    = ...
    DT_LNK     = 10
    DT_REG     = ...
    DT_SOCK    = ...
    DT_UNKNOWN = 0
  end
end
```

especially if the original C source didn't have an enum. It probably does not mean directly embedding the integer constant in the source code.

Reply:
I see, thanks! Fixed.